saw-remote-api/src/SAWServer/Eval.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+module SAWServer.Eval
+  ( evalInt
+  , evalIntDescr
+  , evalBool
+  , evalBoolDescr
+  ) where
+
+import Control.Exception ( throw )
+import Control.Lens ( view )
+import Control.Monad.IO.Class ( MonadIO(liftIO) )
+import Data.Aeson
+    ( (.:),
+      withObject,
+      object,
+      FromJSON(parseJSON),
+      KeyValue((.=)),
+      ToJSON(toJSON) )
+
+import qualified Argo
+import qualified Argo.Doc as Doc
+import CryptolServer.Data.Expression ( Expression(..), getCryptolExpr )
+import Verifier.SAW.TypedTerm (TypedTerm)
+import qualified SAWScript.Builtins as SB
+import qualified SAWScript.Value as SV
+import SAWServer
+    ( SAWState,
+      sawBIC,
+      sawTopLevelRW)
+import SAWServer.CryptolExpression ( CryptolModuleException(..), getTypedTermOfCExp )
+import SAWServer.TopLevel ( tl )
+
+data EvalParams a cryptolExpr =
```

Comment:
Since `a` is a phantom type, it's perhaps not obvious at a first glance that this is meant to encode the type being evaluated to. It would be worth including a Haddock comment to this effect.